.github/workflows/shellcheck.yml

```diff
@@ -0,0 +1,29 @@
+name: ShellCheck
```

Comment:
Would it make sense to name this job and yml file something more generic like `Linting` in case we want to add other tools as additional `steps`? It wouldn't make sense to add more CI jobs for them since this one says `Successful in 5s`.

Reply:
We could do that, but there wouldn't be many advantages imho. Adding short jobs is better for parallelism.

Reply:
There is a finite limit on the number of jobs, correct? That is the dominating factor in my mind. I'm not sure how parallelism comes into play when the Shellcheck step only takes five seconds.

Reply:
Yes, there is a limit of concurrent jobs (20 on GitHub). Note that the check takes 5 secs at the moment because there are very few files checked. Checking all our internal scripts (not including hwloc and romio) as I hope to do in the future will probably take several mins on CI. In that case, running (e.g.) cpplint concurrently makes sense I think.